### PR TITLE
[MU4] Templates repository 

### PR DESCRIPTION
**Pull request by RomanPudashkin**
- implemented templates repository
- fixed tests execution

### CODE REVIEW

File: framework/system/ifsoperations.h [hunk 1]
```diff
@@ -46,6 +46,13 @@ class IFsOperations : MODULE_EXPORT_INTERFACE
 
     virtual RetVal<QStringList> directoryFileList(const QString& path, const QStringList& nameFilters,
                                                   QDir::Filters filters = QDir::NoFilter) const = 0;
+
+    enum class ScanMode {
+        CurrentDirOnly,
+        IncludeSubdirs
+    };
+
+    virtual QStringList scanForFiles(const QString& dirPath, const QStringList& filesFilters, ScanMode mode) const = 0;
```
Comment (Eism) [inline framework/system/ifsoperations.h:55]: Could you explain what is the purpose of this method? And what is the difference between it and directoryFileList?

Comment (RomanPudashkin) [reply framework/system/ifsoperations.h:55]: directoryFileList can return directories also. Plus, this method can scan directories for files recursively. Perhaps better to rename directoryFileList to entryList

File: mu4/domain/notation/internal/templatesrepository.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        QStringList filters { "*.mscz", "*.mscx" };
+        return fsOperations()->scanForFiles(dirPath, filters, IFsOperations::ScanMode::IncludeSubdirs);
+    };
+
+    QStringList result;
+
+    result << scanForTemplates(configuration()->templatesPath());
+    result << scanForTemplates(configuration()->userTemplatesPath());
+
+    for (const QString& extensionTemplatesPath: configuration()->extensionsTemplatesPaths()) {
```
Comment (Eism) [inline mu4/domain/notation/internal/templatesrepository.cpp:39]: Better to do one method in the configuration with getting all available paths.

File: framework/system/ifsoperations.h (hunk 1 above)
Comment (igorkorsukov) [reply framework/system/ifsoperations.h:55]: Why `scanForFiles`, not just `scanFiles`?

Comment (igorkorsukov) [reply framework/system/ifsoperations.h:55]: Let's decide:
One method like: 
`RetVal<QStringList> scanEntry(const QString& path, const QStringList& nameFilters, EntryType type, ScanMode mode)`

where:
EntryType - Dir, Files, All
ScanMode - CurrentDirOnly, IncludeSubdirs

Multiple methods like:
`RetVal<QStringList> scanDirs(const QString& path, const QStringList& nameFilters, ScanMode mode)`
`RetVal<QStringList> scanFiles(const QString& path, const QStringList& nameFilters, ScanMode mode)`

File: mu4/domain/notation/internal/templatesrepository.cpp (hunk 2 above)
Comment (RomanPudashkin) [reply mu4/domain/notation/internal/templatesrepository.cpp]: fixed

File: framework/system/ifsoperations.h (hunk 1 above)
Comment (RomanPudashkin) [reply framework/system/ifsoperations.h]: I have removed directoryFileList and replaced it on scanFiles

File: mu4/userscores/internal/templatesrepository.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#include "templatesrepository.h"
```
Comment (Eism) [inline mu4/userscores/internal/templatesrepository.cpp:1]: Please add license.

Comment (RomanPudashkin) [reply mu4/userscores/internal/templatesrepository.cpp:20]: fixed

**Pull request merged**